zoloto/coords.py

```diff
@@ -57,17 +119,58 @@ def __init__(self, e_x: float, e_y: float, e_z: float):
 
     @property
     def rot_x(self) -> float:
-        """Get rotation angle around x axis in radians."""
+        """
+        Get rotation angle around X axis in radians.
+
+        The X axis is horizontal relative to the camera's perspective, i.e: left
+        & right within the frame of the image.
+
+        Increasing values represent an increasing clockwise rotation of the
+        marker as seen from the camera's left.
+
+        Zero values for April Tags markers have the marker facing away from the
+        camera. The practical effect of this is that an April Tags marker facing
+        the camera square-on will have a value of ``pi`` (or equivalently
+        ``-pi``) and the value will decrease as the marker diverges from
+        square-on.
+
+        For observed markers positive values therefore indicate a rotation of
+        the top of the marker away from the camera, such that marker could be
+        said to be leaning backwards, with the value decreasing as the marker
+        leans back further.
+        """
         return self.roll
 
     @property
     def rot_y(self) -> float:
-        """Get rotation angle around y axis in radians."""
+        """
+        Get rotation angle around Y axis in radians.
+
+        The Y axis is vertical relative to the camera's perspective, i.e: up &
+        down within the frame of the image.
+
+        Positive values indicate a rotation of an observed marker towards the
+        camera's right. This is a rotation of the marker counter-clockwise about
+        the Y axis as seen from above the marker.
+
+        Zero values for April Tags markers have the marker facing the camera
+        square-on.
+        """
         return self.pitch
 
     @property
     def rot_z(self) -> float:
-        """Get rotation angle around z axis in radians."""
+        """
+        Get rotation angle around Z axis in radians.
+
+        The Z axis extends directly away from the camera.
+
+        Positive values indicate a rotation counter-clockwise from the
+        perspective of the camera.
+
+        Zero values for April Tags markers have the marker reference point at
+        the top left.
+        """
         return self.yaw
 
     @property
```

Comment:
These don't really make sense with the current coordinate system.

Reply:
Could you elaborate on what you mean? Both in terms of which version you're referring to as "current" and which of the properties you're referring to here.

Reply:
Yaw, pitch and roll do not line up with their standard definition https://en.wikipedia.org/wiki/Aircraft_principal_axes?wprov=sfla1

For instance yaw is a rotation about the vertical axis which is not the case here.

Reply:
I would probably say that the correct fix here is to fix the incorrect convention, as we discussed in the kit team meeting last night.

Reply:
Unfortunately changing the mappings of which axes are which has the potential to be severely breaking to competitors code.
Even the changes which are in this branch as it stands are technically breaking, though relatively unlikely to affect most cases (and have the advantage of making this logic match what's actually in the SR docs).